test/Microsoft.AspNet.Routing.Tests/Template/TemplateRouteTest.cs

```diff
@@ -971,6 +973,86 @@ public void GetVirtualPath_Fail()
             Assert.Null(path);
         }
 
+        [Theory]
+        [MemberData("DataTokensTestData")]
+        public void GetVirtualPath_ReturnsDataTokensWhenTargetReturnsVirtualPathData(
+            RouteValueDictionary dataTokens)
+        {
+            // Arrange
+            var path = "TestPath";
+
+            var target = new Mock<IRouter>(MockBehavior.Strict);
+            target
+                .Setup(r => r.GetVirtualPath(It.IsAny<VirtualPathContext>()))
+                .Callback<VirtualPathContext>(c => c.IsBound = true)
+                .Returns(() => new VirtualPathData(target.Object, path, dataTokens));
+
+            var route = CreateRoute(
+                target.Object,
+                "{controller}",
+                defaults: null,
+                dataTokens: dataTokens);
```

Comment:
In this test you have two routes that both share the data tokens.
- If you intended to test the 'merge' scenario then you need to have different data so you can really see that it works. 
- Otherwise please clean this up so we can see what the test intends to cover